core/controllers/platform_feature_test.py

```diff
@@ -0,0 +1,147 @@
+        )
+
+        feature_services.ALL_FEATURES_LIST = param_names
+        feature_services.ALL_FEATURES_NAMES_SET = set(param_names)
+
+    def tearDown(self):
+        super(PlatformFeaturesEvaluationHandlerTest, self).tearDown()
+
+        feature_services.ALL_FEATURES_LIST = self.original_feature_list
+        feature_services.ALL_FEATURES_NAMES_SET = self.original_feature_name_set
+        registry.Registry.parameter_registry = self.original_registry
+
+    def test_get_dev_mode_android_client_returns_correct_flag_values(self):
+        with self.swap(constants, 'DEV_MODE', True):
+            result = self.get_json(
+                '/platform_features_evaluation_handler',
+                params={
+                    'client_type': 'Android',
+                    'app_version': '1.0.0',
+                    'user_locale': 'en',
+                }
+            )
+            self.assertEqual(
+                result,
+                {self.dev_feature.name: False, self.prod_feature.name: True})
+
+    def test_get_with_invalid_client_type_context_raises_400(self):
+        resp_dict = self.get_json(
+            '/platform_features_evaluation_handler',
+            params={
+                'client_type': 'Invalid',
+                'app_version': '1.0.0',
+                'user_locale': 'en',
+            },
+            expected_status_int=400
+        )
+        self.assertEqual(
+            resp_dict['error'],
+            'Invalid client type \'Invalid\', must be one of [u\'Web\', '
+            'u\'Android\'].'
+        )
+
+
+class PlatformFeatureDummyHandlerTest(test_utils.GenericTestBase):
+    """Tests for the PlatformFeatureDummyHandler."""
+
+    def setUp(self):
+        super(PlatformFeatureDummyHandlerTest, self).setUp()
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.user_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        self._enable_dummy_feature()
+
+    def _enable_dummy_feature(self):
+        """Enables the dummy_feature for the dev environment."""
+        feature_services.update_feature_flag_rules(
+            param_list.PARAM_NAMES.dummy_feature, self.user_id,
+            'update rule for testing purpose',
+            [{
+                'value_when_matched': True,
+                'filters': [{
+                    'type': 'server_mode',
+                    'conditions': [['=', param_domain.SERVER_MODES.dev]]
+                }]
+            }]
+        )
+
+    def test_get_with_dummy_feature_enabled_returns_ok(self):
+        with self.swap(constants, 'DEV_MODE', True):
+            result = self.get_json(
+                '/platform_feature_dummy_handler',
+            )
+            self.assertEqual(result, {'msg': 'ok'})
+
+    def test_get_with_dummy_feature_disabled_raises_404(self):
+        with self.swap(constants, 'DEV_MODE', False):
+            self.get_json(
+                '/platform_feature_dummy_handler',
+                expected_status_int=404
```

Comment:
These tests seem to differ in the server mode, not the enabled/disabled state of the feature. I suggest also adding a test for the feature being off (but in the correct server environment).